test/cmocka/src/math/matrix/ref_matrix_mult_16_test2.h

```diff
@@ -0,0 +1,172 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+/* This file was created with script ref_matrix.m */
+
+#define MATRIX_MULT_16_TEST2_ELEMENTWISE  0
+#define MATRIX_MULT_16_TEST2_A_ROWS  6
+#define MATRIX_MULT_16_TEST2_A_COLUMNS  7
+#define MATRIX_MULT_16_TEST2_B_ROWS  7
+#define MATRIX_MULT_16_TEST2_B_COLUMNS  8
+#define MATRIX_MULT_16_TEST2_C_ROWS  6
+#define MATRIX_MULT_16_TEST2_C_COLUMNS  8
+#define MATRIX_MULT_16_TEST2_A_QXY_Y  0
+#define MATRIX_MULT_16_TEST2_B_QXY_Y  0
+#define MATRIX_MULT_16_TEST2_C_QXY_Y  0
+
+static const int16_t matrix_mult_16_test2_a[42] = {
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+};
+
+static const int16_t matrix_mult_16_test2_b[56] = {
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+	1,
+};
+
+static const int16_t matrix_mult_16_test2_c[48] = {
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
+	7,
```

Comment:
Does this need to use 170 identical lines?

Reply:
It's 100% generated C code. I added a trivial case that is easy to hand verify. It could be a different test though with other Q16.0 random numbers.

Reply:
Well... could it be generated better? :-)

BTW a generation timestamp and `git describe` would not hurt.

Reply:
OK, since you demand now some Q0 random integers instead :)

I added to created comment date in this format "30-Aug-2022 14:01:32". But I don't know how I could utilize git describe. I don't think these test vector files are worth tagging with version. In bisect they can be identified with commit git hash anyway.

Reply:
Moved to https://github.com/thesofproject/sof/pull/6198#issuecomment-1232006486